### PR TITLE
DeleteAll #107

### PR DESCRIPTION
**Pull request by waldo1001**

### CODE REVIEW

File: content/BCBestPractices/Performance/DeleteAll/index.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
++++
+title = "DeleteAll"
+weight = 1180
++++
+This is a guideline, some parts are optional (if there's no content, remove the whole paragraph).
```
Comment (christianbraeunlich) [inline content/BCBestPractices/Performance/DeleteAll/index.md:5]: I don't think this line is needed anymore.

File: content/BCBestPractices/Performance/DeleteAll/index.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+```al
+    EmptyTableWLD.Setrange(Code, 'AJ');
+    EmptyTableWLD.DeleteAll(true);
+```
+
+## Good code
+
+```al
+    EmptyTableWLD.Setrange(Code, 'AJ');
```
Comment (christianbraeunlich) [inline content/BCBestPractices/Performance/DeleteAll/index.md:24]: I would suggest to rename the procedure call to Set**R**ange.

File: content/BCBestPractices/Performance/DeleteAll/index.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+## Description
+
+When you perform a "DeleteAll" when there is nothing to delete, it will still perform a lock.  When you for example perform a DeleteAll on an empty table, it will result in a table lock.
+Therefore it's good practice to always check if the table is empty when performing a DeleteAll.
+
+## Bad code
+
+```al
+    EmptyTableWLD.Setrange(Code, 'AJ');
```
Comment (christianbraeunlich) [inline content/BCBestPractices/Performance/DeleteAll/index.md:17]: I would suggest to rename the procedure call to Set**R**ange.

File: content/BCBestPractices/Performance/DeleteAll/index.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    if not EmptyTableWLD.IsEmpty() then
+        EmptyTableWLD.DeleteAll(true);
+```
+
+## [Discussions](https://github.com/microsoft/alguidelines/discussions/categories/bc-best-practices?discussions_q=DeleteAll+category%3A%22BC+Best+Practices%22)
+
+You can find discussions on all "Best Practices" [here](https://github.com/microsoft/alguidelines/discussions/categories/bc-best-practices).
+
+If you don't find the discussion of this guideline, please feel fee to create a new one with the same title as this article.
```
Comment (christianbraeunlich) [inline content/BCBestPractices/Performance/DeleteAll/index.md:33]: I would suggest to rename fee to f**r**ee.
Will open an Issue for this as this occures in every guidline probably has it's roots in the template.

File: content/BCBestPractices/Performance/DeleteAll/index.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+## Good code
+
+```al
+    EmptyTableWLD.SetRange(Code, 'AJ');
+    if not EmptyTableWLD.IsEmpty() then
+        EmptyTableWLD.DeleteAll(true);
+```
+
+## [Discussions](https://github.com/microsoft/alguidelines/discussions/categories/bc-best-practices?discussions_q=DeleteAll+category%3A%22BC+Best+Practices%22)
```
Comment (TheDoubleH) [inline content/BCBestPractices/Performance/DeleteAll/index.md:28]: Do You really want the discussion link to be a header?

Comment (waldo1001) [reply content/BCBestPractices/Performance/DeleteAll/index.md:28]: Yeah - that doesn't really make sense, I agree .. :(

**Review by JesperSchulz — APPROVED** [review]

**Review by TheDoubleH — APPROVED** [review]

**Review by JeremyVyska — APPROVED** [review]
❤

**Pull request merged**